test/integration/migration.spec.js

```diff
@@ -376,17 +377,10 @@ describe('the migration', function () {
     ]);
   }));
 
-  it('throws an error when the script is invalid', co(function * () {
-    let result;
-
-    try {
-      result = yield migrationParser(invalidScript);
-    } catch (err) {
-      expect(err).to.be.an('error');
-      expect(err.message).to.include('Validation failed');
-    }
+  it('returns an error when the script is invalid', co(function * () {
```

Comment:
I think we should (if we've some time left) do a couple extra things:

- Rename this test to explicitly mention that it's only about `payloadValidationErrors`
- Add some tests for the other possible errors included in the `payloadResult`